catalog/app/containers/Bucket/PackageDialog/FilesInput.tsx

```diff
@@ -246,7 +101,20 @@ const insertIntoTree = (path: string[] = [], file: FilesEntry, entries: FilesEnt
       restEntries = R.without([existingDir], entries)
       baseDir = existingDir as FilesEntryDir
     }
-    inserted = insertIntoDir(rest, file, baseDir)
+    if (
+      FilesEntry.match(
```

Comment:
this condition looks quite arcane, maybe it's worth it to use a variable with some descriptive name

Reply:
Basically, it is "if the file is hidden and if it is the file, not the parent directory". Ended up with:
```tsx
    // If file is "hidden",
    // and it is the actual file, not a parent path;
    // then we skip inserting it into UI
    const hiddenFileBase =
      FilesEntry.match(
        {
          File: (f) => f.type === 'hidden',
          Dir: () => false,
        },
        file,
      ) && !rest.length

```